src/Toast.js

```diff
@@ -0,0 +1,116 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import Fade from './Fade';
+import Header from './ToastHeader';
+import Body from './ToastBody';
+import { createBootstrapComponent } from './ThemeProvider';
+import ToastContext from './ToastContext';
+
+const propTypes = {
+  /**
+   * @default 'toast'
+   */
+  bsPrefix: PropTypes.string,
+
+  /**
+   * Apply a CSS fade transition to the toast
+   */
+  animation: PropTypes.bool,
+
+  /**
+   * Auto hide the toast
+   */
+  autohide: PropTypes.bool,
+
+  /**
+   * Delay hiding the toast (ms)
+   */
+  delay: PropTypes.number,
+
+  /**
+   * A Callback fired when the close button is clicked.
+   */
+  onClose: PropTypes.func,
+
+  /**
+   * When `true` The modal will show itself.
+   */
+  show: PropTypes.bool,
+
+  /**
+   * A `react-transition-group` Transition component used to animate the Toast on dismissal.
+   */
+  transition: PropTypes.elementType,
+
+  /** @ignore */
+  innerRef: PropTypes.any,
+};
+
+const defaultProps = {
+  animation: true,
+  autohide: false,
+  delay: 3000,
+  show: true,
+  transition: Fade,
+};
+
+const Toast = ({
+  bsPrefix,
+  className,
+  children,
+  transition: Transition,
+  show,
+  animation,
+  delay,
+  autohide,
+  onClose,
+  innerRef,
+  ...props
+}) => {
+  useEffect(() => {
+    if (autohide && show) {
```

Comment:
do something like:

```js
const showRef = useRef(false);

useEffect(() => {
  if (autohide && show && !showRef.current) {
    // ...
  }

  // ...
}, [autohide, show]);

showRef.current = show;
```

Reply:
or maybe we don't need `showRef`... but we don't want to set a new timer on changes to other props, i think

Reply:
I tried to e.g. to change the children values each 100ms and added console.log statements to setTimeout and clearTimeout. That worked well as expected.

Reply:
okay, this is probably fine then